### DISCUSSION TITLE
Clarification on the use-case for searchAll and includeEpisodes

### DISCUSSION THREAD

**Discussion by Jerrk** [cross-seed/cross-seed#232] [General]
As i understand it the default behavior of cross-seed is `searchAll` and `includeEpisodes` set to **False**, this means that it will only search for single-video-file torrents such as

    video.ext

    video
     - video.ext

but it would **not** search for torrents like these

    video
     - video1.ext
     - video2.ext

    video
     - video.ext
     - info.nfo

    video
     - video.ext
     - info.nfo
     - subtitles
       - sub.sub
       - sub.idx
       - sub.srt
     - images
       - cover.jpg
       - poster.jpg

Setting `includeEpisodes` to **True** would then search for torrents like these

    video
     - video1.ext
     - video2.ext

So at this point with `includeEpisodes` to **True** and `searchAll` to **False** only these file structures will be searched for

    video.ext

    video
     - video.ext

    video
     - video1.ext
     - video2.ext

To search for anything other than those 3 file structures you have to turn `searchAll` to **True** right?

for instance i have a couple thousand torrents like

    book.pdf

    book
     - book.pdf
     - book.mp3

    book.zip

    book.rar

    book
     - img1.jpg
     - img2.jpg
     - img3.jpg
     - img4.jpg

    book
     - book.exe

you get the jist - these specific torrents i have can only be found in one place so there is no point calling all of my indexers 4,000 times for torrents i know will not be found there.

This might be a bit in the vein of "what is the purpose for the project"?
Cross seeding any and all files, or be focused on video files specifically?
If it's the latter will there be more "inclusive" options in the future such as allowing .nfo files, subtitles, and posters without having to use `searchAll`?

**Comment by mmgoodnow** [comment]
Most of what you laid out was correct, but I do have a couple corrections:

With the default settings:
- pack folders consisting of only video files will work
- Single-file torrents that look like movies will work (if it's named in the standard way)

The pre filtering stage isn't determined by the shape of the torrent content, but by the naming scheme of the files:

https://github.com/mmgoodnow/cross-seed/blob/689b5a382eaa44b540a302b56961a5b676d4c2b9/src/constants.ts#L1-L7

Other than that, there are several questions here, so I'll try to unpack them:

### Why not music, book, game, or software torrents? 

When I started the project, it was just a simple program that I wrote for myself and posted to GitHub. I made those choices because 
I have no experience with music, book, game, or software torrents. I also don't know if the naming schemes for non-video torrents are as standardized. 

### Why not episodes?

On most private trackers, episodes get deleted once a season ends and packs come out. I didn't (and still don't) see the point in cross-seeding episodes which would just increase the number of torrents I'd have to manually delete from my client at the end of a season. 

### Why not nfos, subtitle files, posters, samples, rars, etc?

Because I wanted to cover the 80% case that was easy and more likely to match across multiple trackers. When I created cross-seed, I made an assumption that video torrents with extra files would be less likely to match, so I decided to just filter them out entirely. I also wanted to be very conservative because cross-seeding is pretty heavy on trackers' APIs (and most trackers don't even have an api, they just get web-scraped, which makes it even heavier), so I only wanted to search for the torrents that would be the most likely to have matches. In my eyes, this is sufficient because cross-seeding is an optimization, and I consider _any_ amount of cross-seeds to be a "bonus". 

The other reason is that I don't like when torrents have those extra files. IMO, all movie torrents should be a single file and all subtitles should be embedded. If cross-seed ever gets big enough that it can exert any sort of pressure on release groups to follow a more rigid pattern, I want it to urge people towards single-file movie torrents, no nfos, etc. 

I haven't given a lot of development to the other cases. I added the includeEpisodes and searchAll flags early on because they were easy to implement. At the time, I was also feeling very competitive towards https://github.com/BC44/Cross-Seed-AutoDL so I wanted to make sure I had feature parity.

### will there be more inclusive options in the future such as allowing .nfo files, subtitles, and posters without having to use `searchAll`?

Several people have asked for different prefilters for their use cases, so I created #210. I want to try to provide a solution along the lines of "easy things should be easy, hard things should be possible". My current idea is to allow people to provide a JS function inside their config file to filter. However, I don't plan on adding more specific options as I want to keep the defaults pretty opinionated, and I also want to avoid the proliferation of too many command-line options.

**Reply by SawkeeReemo** [reply]
This is great info! Thank you!
